yt_dlp/extractor/vrt.py

```diff
@@ -1,82 +1,215 @@
+                        'client_id': 'vrtnu-site'
+                    }, data=urlencode_postdata(post_data))
+
+            except ExtractorError as e:
+                if isinstance(e.cause, compat_HTTPError) and e.cause.code == 401:
+                    login_attempt += 1
+                    self.report_warning('Authentication failed')
+                    self._sleep(1, None, msg_template='Waiting for %(timeout)s seconds before trying again')
+                else:
+                    raise e
+            else:
+                break
+
+        self._authenticated = True
+
     def _real_extract(self, url):
-        site, display_id = self._match_valid_url(url).groups()
-        webpage = self._download_webpage(url, display_id)
-        attrs = extract_attributes(self._search_regex(
-            r'(<[^>]+class="vrtvideo( [^"]*)?"[^>]*>)', webpage, 'vrt video'))
-
-        asset_id = attrs['data-video-id']
-        publication_id = attrs.get('data-publication-id')
-        if publication_id:
-            asset_id = publication_id + '$' + asset_id
-        client = attrs.get('data-client-code') or self._CLIENT_MAP[site]
-
-        title = strip_or_none(get_element_by_class(
-            'vrt-title', webpage) or self._html_search_meta(
-            ['og:title', 'twitter:title', 'name'], webpage))
-        description = self._html_search_meta(
-            ['og:description', 'twitter:description', 'description'], webpage)
-        if description == '…':
-            description = None
-        timestamp = unified_timestamp(self._html_search_meta(
-            'article:published_time', webpage))
+        display_id = self._match_id(url)
+
+        model_json = self._download_json(f'{url.strip("/")}.model.json', display_id,
+                                         'Downloading asset JSON', 'Unable to download asset JSON')
+        details = model_json.get('details')
+        actions = details.get('actions')
+        title = details.get('title')
+        episode_publication_id = actions[2].get('episodePublicationId')
+        episode_video_id = actions[2].get('episodeVideoId')
+        video_id = f'{episode_publication_id}${episode_video_id}'
+        description = details.get('description')
+        episode = details.get('data').get('episode')
+        display_id = episode.get('name')
+        timestamp = parse_iso8601(episode.get('onTime').get('raw'))
+        upload_date = strftime('%Y%m%d', gmtime(timestamp))
+        series_info = details.get('data')
+        series = series_info.get('program').get('title')
+        season = series_info.get('season').get('title').get('value')
+        season_number = series_info.get('season').get('title').get('raw')
+        season_id = series_info.get('season').get('id')
+        episode = series_info.get('episode').get('number').get('value')
+        episode_number = series_info.get('episode').get('number').get('raw')
+        episode_id = series_info.get('episode').get('id')
```

Comment:
we can and should inline most of this in the return
```suggestion
        display_id = self._match_id(url)
        details = self._download_json(
            f'{url.strip("/")}.model.json', display_id, 'Downloading asset JSON',
            'Unable to download asset JSON')['details']

        watch_info = traverse_obj(details, (
            'actions', lambda _, v: v['type'] == 'watch-episode', {dict}), get_all=False) or {}
        video_id = join_nonempty(
            'episodePublicationId', 'episodeVideoId', delim='$', from_dict=watch_info)
        if '$' not in video_id:
            raise ExtractorError('Unable to extract video ID')
```